formats/src/main/scala/org/scanamo/DerivedDynamoFormat.scala

```diff
@@ -28,30 +28,28 @@ trait DerivedDynamoFormat {
 
   implicit def hcons[K <: Symbol, V, T <: HList](
     implicit
-    headFormat: Lazy[DynamoFormat[V]],
-    tailFormat: Lazy[ConstructedDynamoFormat[T]],
+    headFormat0: Lazy[DynamoFormat[V]],
+    tailFormat: ConstructedDynamoFormat[T],
     fieldWitness: Witness.Aux[K]
   ): ValidConstructedDynamoFormat[FieldType[K, V] :: T] =
     new ValidConstructedDynamoFormat[FieldType[K, V] :: T] {
-      final private val fieldName = fieldWitness.value.name
+      private[this] val fieldName = fieldWitness.value.name
+      private[this] val headFormat = headFormat0.value
 
-      final def read(av: DynamoObject) = {
-        val valueOrError: Either[(FieldName, DynamoReadError), V] =
-          av(fieldName)
-            .map(headFormat.value.read)
-            .orElse(headFormat.value.default.map(Either.right))
-            .getOrElse(Either.left(MissingProperty))
-            .leftMap(((fieldName, _)))
-
-        val head = valueOrError.toValidatedNel.map(field[K](_))
-        val tail = tailFormat.value.read(av)
+      final def read(av: DynamoObject): ValidatedPropertiesError[FieldType[K, V] :: T] = {
+        val head = av(fieldName)
+          .fold(headFormat.read(DynamoValue.nil).leftMap[DynamoReadError](_ => MissingProperty))(headFormat.read(_))
```

Comment:
For reviewers: this is where I am passing `nil` explicitly if the key is missing from the object. If you take a look at the formats for which there should be a default value, they always test if the value is null and return the default value in that case. Two things:

1. `apply` returning an `Option[DynamoValue]` allows me to isolate the case of a `MissingProperty`
2. OTOH knowing that dynamo will never return a null value, if `apply` returns `DynamoValue` instead, I can infer the `MissingProperty` error if `av(fieldName).isNull`